obspy/io/resp/nrl.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+'''
+nrl.py for navigating, and picking responses from the
+Nominal Response Library.
+
+:copyright:
+    Lloyd Carothers IRIS/PASSCAL, 2016
+:license:
+    GNU Lesser General Public License, Version 3
+    (http://www.gnu.org/copyleft/lesser.html)
+'''
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+# Urllib for py 2 and 3
+from future.standard_library import install_aliases
+install_aliases()  # NOQA
+from urllib.parse import urljoin
+from urllib.request import urlopen
+from os.path import join, dirname
+from ConfigParser import SafeConfigParser
+
+
+class NRL:
+    """
+    Object representing the Nominal Response library.
+    Can be Accessed online from the DMC or with a local copy
+    Decision tree: Usually!!!! depth of tree and order can change!
+    Sensor, make, model, period, sensitivity
+    Datalogger, make, model, gain, sample rate
+
+    The NRL has txt desision files formatted as windows ini
+
+    """
+    URLroot = 'http://ds.iris.edu/NRL/'
+    index = 'index.txt'
+    # Placholder for sample rate and gain
+    SR = object()
+    GAIN = object()
+    dl_shortcuts = {
+        'rt130': ['REF TEK', 'RT 130 & 130-SMA', GAIN, SR],
+        'q330': ['Quanterra', 'Q330SR', GAIN, SR, 'LINEAR AT ALL SPS']}
+    sensor_shortcuts = {
+        'cmg3t': ['Guralp', 'CMG-3T', '120s - 50 Hz', '1500'],
+        'trillium_240_1': [
+            'Nanometrics', 'Trillium 240', '1 - serial numbers < 400'],
+        'l22': ['Sercel/Mark Products', 'L-22D', '5470 Ohms', '20000 Ohms'],
+        'sts2_g3': [
+            'Streckeisen', 'STS-2', '1500', '3 - installed 04/97 to present']
+        }
+
+    def __init__(self, local=True, root=None):
```

Comment:
These two arguments have a strong redundancy in my opinion. I don't see any automatic caching of the NRL done with the `local` keyword right now, I think it is only used to point to a separately downloaded NRL archive on disk somewhere, right? If that's correct, I would get rid of the `local` kwarg and just use `def __init__(self, root='http://ds.iris.edu/NRL/')`, the user could then use a local archive with e.g. `NRL(root='/some/path/to/NRL')`. In the code you could do a check like [this one](https://github.com/obspy/obspy/blob/master/obspy/core/stream.py#L220) and then decide whether to do local filesystem parsing or URL fetching.

Also, a second thought: Maybe it would even be cleaner to have a class hierarchy with..

```python
class NRL(object):
    # common methods here
    def __init__(self, root='http://ds.iris.edu/NRL'):
        if "://" in root:
            self = RemoteNRL(root=root)
        else:
            self = LocalNRL(root=root)
```

.. and then have the routines like `read_ini` and so on with same names in both subclasses? I think this would be a bit cleaner and clearer..